lib/sparoid.rb

```diff
@@ -149,9 +149,16 @@ def write_cache
     end
   end
 
-  def public_ip
-    Resolv::DNS.open(nameserver: ["208.67.222.222", "208.67.220.220"]) do |dns|
-      dns.getresource("myip.opendns.com", Resolv::DNS::Resource::IN::A).address
+  def public_ip(host = "checkip.amazonaws.com")
+    Socket.tcp(host, 80, connect_timeout: 3) do |sock|
+      sock.sync = true
+      sock.print "GET / HTTP/1.1\r\nHost: #{host}\r\nConnection: close\r\n\r\n"
+      status = sock.readline(chomp: true)
+      raise ResolvError.new("#{host} response: #{status}") unless status.start_with? "HTTP/1.1 200 "
+      until sock.readline(chomp: true).empty? 
+        # skip all headers
+      end
+      sock.readline(chomp: true) # this is the IP
```

Comment:
```suggestion
      ip = sock.readline(chomp: true) # this is the IP
      Resolv::IPv4.create(ip)
```

To keep return value same? Otherwise I think we gotta look at `#message` to generate binary output

Reply:
did https://github.com/84codes/sparoid.rb/pull/14/commits/2b07eb3aa8e28a018c117ee2350ccc844f1a3305

Reply:
Hmm, fails as it is trying to use ruby 2.5 parser, do we really need to build for ruby `< 3`?

Reply:
doesn't hurt to support older ruby versions i guess, it was just the keyword argument passing in a test that wasn't support in ruby < 3.0, changed that.

Reply:
thank you btw! you were very correct on the return type